src/iaphandler.cpp

```diff
@@ -0,0 +1,282 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "iaphandler.h"
+#include "constants.h"
+#include "leakdetector.h"
+#include "logger.h"
+
+#include <QCoreApplication>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonValue>
+#include <QScopeGuard>
+
+#ifdef MVPN_IOS
+#  include "platforms/ios/iosiaphandler.h"
+#else
+#  include "platforms/dummy/dummyiaphandler.h"
+#endif
+
+namespace {
+Logger logger(LOG_IAP, "IAPHandler");
+}  // namespace
+
+// static
+IAPHandler* IAPHandler::createInstance() {
+#ifdef MVPN_IOS
+  return IOSIAPHandler::createInstance();
+#else
+  return DummyIAPHandler::createInstance();
+#endif
+}
+
+// static
+IAPHandler* IAPHandler::instance() {
+#ifdef MVPN_IOS
+  return IOSIAPHandler::instance();
```

Comment:
What about if here you do: `return new IOSIAPHandler();` and `return new DummyIAPHandler()`. ?